### PR TITLE
feat: Add copy task id to task list

### PR DESCRIPTION
**Pull request by gt2345**
## Ticket

[MD-460]

## Description

Allow user to copy task ID from task table action.
Screenshot 2024-10-17 at 11 41 43 AM

## Test Plan

At the task list page, click on the triple dots to expand the action.
Click on copy task ID, and verify that the task ID has been copied to clipboard.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[MD-460]: https://hpe-aiatscale.atlassian.net/browse/MD-460?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JComins000 — APPROVED** [review]
😎

File: webui/react/src/e2e/tests/workspaceTasks.spec.ts [hunk 1]
```diff
@@ -22,13 +22,19 @@ test.describe('Workspace Tasks', () => {
       await firstRow.pwLocator.waitFor({ timeout: 10_000 });
     });
 
-    await test.step('Kill task', async () => {
+    await test.step('Kill task and copy task ID', async () => {
```
Comment (johnkim-det) [inline webui/react/src/e2e/tests/workspaceTasks.spec.ts:25]: nit: separate step for copy task ID?

File: webui/react/src/components/TaskActionDropdown.tsx [hunk 2]
```diff
@@ -97,6 +104,13 @@ const TaskActionDropdown: React.FC<Props> = ({ task, onComplete, children }: Pro
           onComplete?.(key);
           navigate(paths.taskLogs(task));
           break;
+        case Action.CopyTaskID:
+          await copyToClipboard(task.id);
+          openToast({
+            severity: 'Confirm',
+            title: 'Task id has been copied to clipboard.',
```
Comment (johnkim-det) [inline webui/react/src/components/TaskActionDropdown.tsx:111]: nit: caps "ID" is consistent with menu item label.

File: webui/react/src/e2e/tests/workspaceTasks.spec.ts [hunk 3]
```diff
@@ -22,13 +22,19 @@ test.describe('Workspace Tasks', () => {
...
       await workspaceDetails.taskList.taskKillModal.killButton.pwLocator.click();
-
       await expect(firstRow.state.pwLocator).toHaveText('Terminated');
+
+      await context.grantPermissions(['clipboard-read', 'clipboard-write']);
+      await (await firstRow.actions.open()).copy.pwLocator.click();
+      const handle = await authedPage.evaluateHandle(() => navigator.clipboard.readText());
+      const clipboard = await handle.jsonValue();
+      expect(clipboard.split('-').length).toBeGreaterThan(2);
+      await expect(firstRow.taskID.pwLocator).toContainText(clipboard.split('-')[0]);
```
Comment (johnkim-det) [inline webui/react/src/e2e/tests/workspaceTasks.spec.ts:37]: nit: I think you could just render and validate the full id here, but no big deal if that's not feasible. but I do think uuid format is predictable enough to be more precise than number of segments > 2, or could use `validate` function from `uuid` library we're already using in the app.

**Review by johnkim-det — APPROVED** [review]

**Review by azhou-determined — APPROVED** [review]
nice! thanks for doing this

**Pull request merged**